### PR TITLE
Node.js CI, GitHub CodeQL, and Issue/PR management using GitHub Actions

### PR DESCRIPTION
**Pull request by baradghimire**
# Description

Adds the following 4 GitHub Actions workflows:

1. `nodejs-ci.yml` does a clean install of node dependencies, build the source code and run tests across different versions of node, on every push and pull request.
2. `codeql-analysis.yml` Automatic code scanning for common vulnerabilities and coding errors using GitHub CodeQL
3. `greetings.yml` Greets new users when they make their first Issue or PR
4. `stale.yml` Labels stale issues and PRs

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

The workflows were ran on [OnTrack-UG-Squad/doubtfire-web](https://github.com/OnTrack-UG-Squad/doubtfire-web/actions) and confirmed to be working.

## Testing Checklist:

- [x] Tested in OnTrack-UG-Squad/doubtfire-web

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have requested a review from @macite and @jakerenzella on the Pull Request (unable to add Reviewers since I don't have collaborator access)

### CODE REVIEW

**Comment by macite** [comment]
Need to add in the `ng lint` action to ensure code is lint free :)

File: .github/workflows/nodejs-ci.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      matrix:
+        node-version: [10.x, 12.x, 14.x]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - run: npm ci
+      - run: npm run build --if-present
```
Comment (baradghimire) [inline .github/workflows/nodejs-ci.yml:24]:
```suggestion
      - run: ng lint
      - run: npm run build --if-present
```

**Review by baradghimire — COMMENTED** [review]
Thanks @macite, I added the ng lint, all good to go now :)

**Comment by baradghimire** [comment]
- I updated the workflow to install the `ng` CLI globally
- The `ng lint` is now working as expected in the [downstream repo](https://github.com/OnTrack-UG-Squad/doubtfire-web/actions/runs/574443464)

**Comment by baradghimire** [comment]
@macite I have removed the CodeQL ​workflow for now as discussed during our last chat on MS Teams.

Node.js CI is working fine now: https://github.com/OnTrack-UG-Squad/doubtfire-web/runs/2048621604

I hope future squad members and/or other contributors will look into CodeQL further

**Comment by jakerenzella** [comment]
@baradghimire Looks good, is this ready for a review?

**Comment by baradghimire** [comment]
@jakerenzella Yep! It's ready for a review. 

You can check out the workflows running in the downstream repo Actions tab here: https://github.com/OnTrack-UG-Squad/doubtfire-web/actions

**Pull request merged**